File: Algorithm/链表/ListNode.m
//
//  ListNode.m
//  Algorithm
//
//  Created by penggongxu on 2020/6/9.
//  Copyright © 2020 personnal. All rights reserved.
//

#import "ListNode.h"

@implementation ListNode


- (instancetype)initNodeWithValue:(int)value {
    if (self = [super init]) {
        self.value = value;
    }
    return self;
}

+ (instancetype)nodeWithValue:(int)value {
    return [[self alloc] initNodeWithValue:value];
}

@end

File: Algorithm/algorithm/ListNode.swift
//
//  ListNode.swift
//  algorithm
//
//  Created by penggongxu on 2021/3/26.
//  单向链表结点

import Foundation

public class ListNode {
    var value: Int
    var nextNode: ListNode?
    init(value: Int, nextNode: ListNode?) {
        self.value = value
        self.nextNode = nextNode
    }
}
